tern/classes/image.py

```diff
@@ -16,6 +16,8 @@ class Image:
         manifest: the json object representing the image manifest
         config: the image config metadata
         layers: list of layer objects in the image
+        until_layer: image should be analyzed only up to this layer
```

Comment:
I'm not sure this property needs to/should be part of an image object. The `image` object attributes are used to represent the image provided and `until_layer` isn't really describing or storing information about the image object itself. Rather, it is a value related specifically to the output report, dictated by the user. I think we should be able to just use the command line `args` to get this information for the report.

Reply:
1. Initially this was pushed through functions - analysis, generate, etc. Prospector complained about too many arguments thus I've moved it to the Image class that is passed there 
2. In my opinion it does provide valuable information about a partially loaded image (see the L155 in tern/classes/docker_image.py)
If I go back there will be prospector complaining about too many args in functions. I think it could be moved to docker image class but I've seen some issues about analysis on layer-by-layer basis for generic OCI images thus figured it might be usefull there.

2. I pushed it earlier as WIP commit and asked about moving this stuff to the image class, as there was no feedback regarding that I split the commit & rebased on previous ones for clear history

Reply:
Maybe I should adjust the comment to `image should be loaded up to this layer` as that is it's purpose here (while it is used further to limit the analysis too but that depends on len(layers)

Reply:
Let me take a look at reducing the complexity of `analyze_subsequent_layers` because I think using args for this will be simpler and more maintainable in the future.

Reply:
I apologize for missing the image class discussion in your last WIP PR. I think when I was reviewing it I wasn't sure how much time you were going to have to work on wasn't expecting your final PR this week.

Reply:
No worries- it happens to the best :)

Reply:
perhaps `load_until_layer` is more clear when reading through the code that uses this variable?